src/DDTrace/Integrations/Mongo/MongoCollectionIntegration.php

```diff
@@ -19,17 +19,20 @@ protected static function loadIntegration()
 
         // MongoCollection::__construct ( MongoDB $db , string $name )
         self::traceMethod('__construct', function (Span $span, array $args) {
+            $span->setIntegration(MongoIntegration::getInstance());
```

Comment:
Would it be possible to override self::traceMethod to automatically call ->setIntegration to every span ?

Reply:
Very good idea, but I will push to the end of the signature to avoid breaking changes if a user implemented an integration relying on this utility method.